### PR TITLE
Add metadata to enable moving nodes

### LINKED ISSUE #443: Metadata
* **Do you want to request a *feature* or report a *bug*?**
Feature

* **What is the current behavior?**
SVG does not include metadata

* **What is the expected behavior?**
Generated SVG comes with metadata (a priori internal with metadata tag, or otherwise external), in particular to:
     - link the nodes id in the SVG with the corresponding equipment id
     - link edges with the corresponding nodes

* **What is the motivation / use case for changing the behavior?**
Callbacks to interact with the SVG diagram

### PR DESCRIPTION
**Pull request by flo-dup**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?**
Fixes #443

**Other information**:
PR transferred from powsybl/powsybl-network-area-diagram#87

### CODE REVIEW

**Comment by zamarrenolm** [comment]
I think we can review and eventually merge these changes.

They provide basic capabilities for updating the diagram relying only on the metadata, as demonstrated in https://github.com/powsybl/powsybl-dev-tools/pull/59 (continuous update while dragging a node).

We can deal in separate PRs with the rest of modifications commented in:
https://github.com/powsybl/powsybl-diagram/issues/443#issuecomment-1318325892

File: network-area-diagram/src/main/java/com/powsybl/nad/svg/SvgWriter.java [hunk 1]
```diff
@@ -59,6 +59,8 @@ public class SvgWriter {
     private static final String Y_ATTRIBUTE = "y";
     private static final String DY_ATTRIBUTE = "dy";
     private static final String POINTS_ATTRIBUTE = "points";
+    private static final Map<BranchEdge.Side, Integer> SIDE_END =
+            new EnumMap<>(Map.of(BranchEdge.Side.ONE, 1, BranchEdge.Side.TWO, 2));
```
Comment (flo-dup) [inline network-area-diagram/src/main/java/com/powsybl/nad/svg/SvgWriter.java:63]: rather add a `BranchEdge.Side::getNum` which returns `ordinal() + 1`

File: network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/DiagramMetadata.java [hunk 2]
```diff
@@ -106,7 +106,7 @@ public void addNode(String svgId, String equipmentId, String positionX, String p
         nodesMetadata.add(new NodeMetadata(svgId, equipmentId, positionX, positionY));
     }
 
-    public void addEdge(String svgId, String equipmentId) {
-        edgesMetadata.add(new EdgeMetadata(svgId, equipmentId));
+    public void addEdge(String svgId, String equipmentId, String node1DiagramId, String node2DiagramId) {
```
Comment (flo-dup) [inline network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/DiagramMetadata.java:109]: You could now rename the `diagramId`s with `svgId`

File: network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/EdgeMetadata.java [hunk 3]
```diff
@@ -7,34 +7,50 @@
...
  */
 public class EdgeMetadata extends AbstractMetadataItem {
 
     private static final String ELEMENT_NAME = "edge";
+    private static final String NODE1_ATTRIBUTE = "node1";
+    private static final String NODE2_ATTRIBUTE = "node2";
 
-    public EdgeMetadata(String svgId, String equipmentId) {
+    private final String node1DiagramId;
+    private final String node2DiagramId;
```
Comment (flo-dup) [inline network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/EdgeMetadata.java:24]: Same here.

File: network-area-diagram/src/main/java/com/powsybl/nad/svg/SvgWriter.java (hunk 1 above)
Comment (zamarrenolm) [reply network-area-diagram/src/main/java/com/powsybl/nad/svg/SvgWriter.java]: done

File: network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/DiagramMetadata.java (hunk 2 above)
Comment (zamarrenolm) [reply network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/DiagramMetadata.java]: renamed

File: network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/EdgeMetadata.java (hunk 3 above)
Comment (zamarrenolm) [reply network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/EdgeMetadata.java]: attributes and parameters renamed

**Pull request merged**